### PR TITLE
Fix blankposting and doubleposting for 2.8.5

### LINKED ISSUE #113: Need to implement checks to prevent doubleposting
Previously the checks happened on the clientside, but this is being removed in [2.8.5](https://github.com/AttorneyOnline/AO2-Client/pull/246)
This should be checked on the serverside

### PR DESCRIPTION
**Pull request by Crystalwarrior**
fixes https://github.com/AttorneyOnline/tsuserver3/issues/113

### CODE REVIEW

**Comment by mposs00** [comment]
lgtm

File: server/network/aoprotocol.py [hunk 1]
```diff
@@ -534,7 +529,12 @@ def net_cmd_ms(self, args):
 
         if len(text) > max_char:
             return
-            
+
+        # Really simple spam protection that functions on the clientside pre-2.8.5, and really should've been serverside from the start
+        if text.strip() != '' and self.client.area.last_ic_message != None and cid == self.client.area.last_ic_message[8] and text == self.client.area.last_ic_message[4]:
```
Comment (oldmud0) [inline server/network/aoprotocol.py:534]: This will suffer an `AttributeError` because `last_ic_message` does not exist.

Comment (ghost) [reply server/network/aoprotocol.py:534]: I can confirm it is indeed broken. This is the output I get when I try to type in ic.

`[asyncio] base_events@1707 : Exception in callback _ProactorReadPipeTransport._loop_reading(<_OverlappedF...rmal^#0#||#%'>)
handle: <Handle _ProactorReadPipeTransport._loop_reading(<_OverlappedF...rmal^#0#||#%'>)>
Traceback (most recent call last):
  File "C:\Program Files\Python38\lib\asyncio\events.py", line 81, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Program Files\Python38\lib\asyncio\proactor_events.py", line 318, in _loop_reading
    self._data_received(data)
  File "C:\Program Files\Python38\lib\asyncio\proactor_events.py", line 269, in _data_received
    self._protocol.data_received(data)
  File "C:\test\server\network\aoprotocol.py", line 113, in data_received
    self.net_cmd_dispatcher[cmd](self, args)
  File "C:\test\server\network\aoprotocol.py", line 537, in net_cmd_ms
    if text.strip() != '' and self.client.area.last_ic_message != None and cid == self.client.area.last_ic_message[8] and text == self.client.area.last_ic_message[4]:
AttributeError: 'Area' object has no attribute 'last_ic_message'`

Comment (oldmud0) [reply server/network/aoprotocol.py:534]: Fixed in https://github.com/AttorneyOnline/tsuserver3/commit/68b872bfe2e90f15ea05fd74587aaaf34576b0c1.

**Pull request merged**